src/deadline/cinema4d_submitter/font_utils.py

```diff
@@ -0,0 +1,446 @@
+
+        # Extract standard font names
+        try:
+            if len(names_table) > TTF_FAMILY_NAME:
+                metadata["family_name"] = str(names_table[TTF_FAMILY_NAME])
+        except (IndexError, Exception) as e:
+            logger.debug(f"Could not extract family name from {font_path}: {e}")
+
+        try:
+            if len(names_table) > TTF_STYLE:
+                metadata["style"] = str(names_table[TTF_STYLE])
+        except (IndexError, Exception) as e:
+            logger.debug(f"Could not extract style from {font_path}: {e}")
+
+        try:
+            if len(names_table) > TTF_FULL_NAME:
+                metadata["full_name"] = str(names_table[TTF_FULL_NAME])
+        except (IndexError, Exception) as e:
+            logger.debug(f"Could not extract full name from {font_path}: {e}")
+
+        try:
+            if len(names_table) > TTF_POSTSCRIPT_NAME:
+                metadata["postscript_name"] = str(names_table[TTF_POSTSCRIPT_NAME])
+        except (IndexError, Exception) as e:
+            logger.debug(f"Could not extract PostScript name from {font_path}: {e}")
+
+        return metadata
+
+    except Exception as e:
+        logger.debug(f"Failed to extract metadata from {font_path}: {e}")
+        return None
+
+
+def is_asset_a_font(asset: dict) -> bool:
+    """
+    Check if Cinema 4D considers the asset is a font.
+    """
+    pid: int = asset.get("paramId", c4d.NOTOK)
+    owner: Optional[c4d.BaseList2D] = asset.get("owner", None)
+    if pid == c4d.NOTOK or owner is None:
+        return False
+    # We check the type of the parameter value.
+    # If it is a FontData, we assume it is a font asset.
+    value = owner.GetParameter(pid, c4d.DESCFLAGS_GET_NONE)
+    return isinstance(value, c4d.FontData)
+
+
+def get_font_location(font_name: str) -> Optional[str]:
+    """
+    Scan font directories for Windows and Mac to find the location of a font.
+
+    Args:
+        font_name (str): Name of the font to find
+
+    Returns:
+        Optional[str]: Path to the font file if found, None otherwise
+    """
+    if not font_name or not font_name.strip():
+        return None
+
+    font_name = font_name.strip()
+    font_dirs = get_system_font_directories()
+
+    for font_dir in font_dirs:
+        try:
+            for root, _, files in os.walk(font_dir):
+                for file in files:
+                    file_path = os.path.join(root, file)
+
+                    # Check if it's a font file by extension and exists
+                    if not is_font_file(file_path):
+                        continue
+
+                    # Check if this font matches
+                    if _is_font_match(font_name, file, file_path):
+                        logger.debug(f"Font match found: {file_path}")
+                        return file_path
+
+        except (PermissionError, OSError) as e:
+            logger.debug(f"Cannot access font directory {font_dir}: {e}")
```

Comment:
There are a lot of places where we log a debug message if there's an exception. How do these debug messages show up to users? If there's an error and a font isn't attached, is there a way for a user to figure out why?